openedx/core/djangoapps/programs/utils.py

```diff
@@ -90,6 +91,9 @@ def __init__(self, site, user, enrollments=None, uuid=None):
             # We can't use dict.keys() for this because the course run ids need to be ordered
             self.course_run_ids.append(enrollment_id)
 
+        self.entitlements = entitlements or list(CourseEntitlement.entitlements_for_user(self.user))
```

Comment:
Question here about explicitly passing falsy values to `entitlements`. 

Is this meant to capture the case where we are using the default value of `None` and look up the entitlements in that case
OR
Do we always want to look up entitlements if a falsy value is passed E.G. If I pass an empty list

Reply:
That is a good question. I modeled this line after how we initialize enrollments for a ProgramProgressMeter above, but it does seem to be a faulty method.